manim/utils/hashing.py

```diff
@@ -30,56 +32,127 @@ def default(self, obj):
             Python object that JSON encoder will recognize
 
         """
-        if inspect.isfunction(obj) and not isinstance(obj, ModuleType):
+        if not (isinstance(obj, ModuleType)) and isinstance(
+            obj, (MethodType, FunctionType)
+        ):
             cvars = inspect.getclosurevars(obj)
             cvardict = {**copy.copy(cvars.globals), **copy.copy(cvars.nonlocals)}
             for i in list(cvardict):
                 # NOTE : All module types objects are removed, because otherwise it throws ValueError: Circular reference detected if not. TODO
                 if isinstance(cvardict[i], ModuleType):
                     del cvardict[i]
-            return {"code": inspect.getsource(obj), "nonlocals": cvardict}
+            return self._check_iterable(
+                {"code": inspect.getsource(obj), "nonlocals": cvardict}
+            )
         elif isinstance(obj, np.ndarray):
-            return list(obj)
+            if obj.size > 1000:
+                obj = np.resize(obj, (100, 100))
+                return "TRUNCATED ARRAY: " + repr(obj)
+            # We return the repr and not a list to avoid the JsonEncoder to iterate over it.
+            return repr(obj)
         elif hasattr(obj, "__dict__"):
             temp = getattr(obj, "__dict__")
-            return self._encode_dict(temp)
+            # MappingProxy is not supported by the Json Encoder
+            if isinstance(temp, MappingProxyType):
+                return dict(temp)
+            return self._check_iterable(temp)
         elif isinstance(obj, np.uint8):
             return int(obj)
-        try:
-            return json.JSONEncoder.default(self, obj)
-        except TypeError:
-            # This is used when the user enters an unknown type in CONFIG. Rather than throwing an error, we transform
-            # it into a string "Unsupported type for hashing" so that it won't affect the hash.
-            return "Unsupported type for hashing"
-
-    def _encode_dict(self, obj):
-        """Clean dicts to be serialized : As dict keys must be of the type (str, int, float, bool), we have to change them when they are not of the right type.
-        To do that, if one is not of the good type we turn it into its hash using the same
-        method as all the objects here.
+        return "Unsupported type for serializing ->" + str(type(obj))
```

Comment:
Format, also missing space after arrow
```suggestion

        return f"Unsupported type for serializing -> {str(type(obj))}"
```